City-of-Helsinki/drupal-helfi-rekry
# UHF-9069 job whitescreen

# [UHF-9069](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9069) 
(fyi: wrong branch name)
Youtube links with bad query parameters can cause trouble while rendering the job listings.

## What was done
Added the same validation used in form field validator to prevent bad links being migrated.
Added query parameter removal if link just happens to have bad query parameters.

`How to recreate:`
- Before installing the fix, try running the validation function with drush
`drush php-eval "_helfi_rekry_content_get_video_url('www.youtube.com/watch?app=desktop&v=lN3Uv8xmAXc')"`
You should not see any exceptions or errors. It means that the link would be saved in database and cause error later on. If you tried this link through remote video field in drupal UI, you would get an error.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-6069_job_whitescreen`
  * `make fresh`
* Run `make drush-cr`

## How to test
Run this command:
`drush php-eval "_helfi_rekry_content_get_video_url('www.youtube.com/watch?app=desktop&v=lN3Uv8xmAXc')"`
- You should see a message telling you that the url has been changed.
- Test the drush command with different urls mentioned in jira-ticket comment. 
- Broken links should show Drupal\migrate\MigrateSkipRowException
- Fixable links should be fixed and notice should be seen. Notice should contain a link which can be tested again and it should work (fixable link = youtube link with bad query parameters, for example &foo=bar)
- Links with wrong provider should show the exception as well (facebook for example)



[UHF-6069]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-6069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UHF-9069]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
